src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -0,0 +1,247 @@
+                {
+                    result.Succeeded = false;
+                    result.ErrorMessage = SR.ProfilerConnectionNotFound;
+                }
+
+                await requestContext.SendResult(result);
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
+        }
+
+        /// <summary>
+        /// Handle request to stop a profiling session
+        /// </summary>
+        internal async Task HandleStopProfilingRequest(StopProfilingParams parameters, RequestContext<StopProfilingResult> requestContext)
+        {
+            try
+            {
+                monitor.StopMonitoringSession(parameters.SessionId);
+                await requestContext.SendResult(new StopProfilingResult());
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            } 
+        }
+
+        /// <summary>
+        /// Starts a new profiler session for the provided connection
+        /// </summary>
+        internal ProfilerSession StartSession(ConnectionInfo connInfo)
+        {
+            // create a new XEvent session and Profiler session
+            var xeSession = this.XEventSessionFactory.CreateXEventSession(connInfo);
+            var profilerSession = new ProfilerSession()
+            {
+                SessionId = Guid.NewGuid().ToString(),
+                XEventSession = xeSession
+            };
+
+            // start monitoring the profiler session
+            monitor.StartMonitoringSession(profilerSession);
+
+            return profilerSession;
+        }
+
+        /// <summary>
+        /// Create a new XEvent sessions per the IXEventSessionFactory contract
+        /// </summary>
+        public IXEventSession CreateXEventSession(ConnectionInfo connInfo)
+        {
+            var sqlConnection = ConnectionService.OpenSqlConnection(connInfo);
+            SqlStoreConnection connection = new SqlStoreConnection(sqlConnection);
+            Session session = ProfilerService.GetOrCreateSession(connection, "Profiler");
+
+            // create xevent session wrapper
+            return new XEventSession()
+            {
+                Session = session
+            };
+        }
+
+        /// <summary>
+        /// Gets an existing XEvent session or creates one if no matching session exists.
+        /// Also starts the session if it isn't currently running
+        /// </summary>
+        private static Session GetOrCreateSession(SqlStoreConnection connection, string sessionName)
+        {
+            XEStore store = new XEStore(connection);
+            Session session = store.Sessions["Profiler"];
+            try
+            {
+                // start the session if it isn't already running
+                if (session != null && !session.IsRunning)
+                {
+                    session.Start();
+                }
+            }
+            catch { }
```

Comment:
this should either throw the exception or create session with error message, otherwise client won't know why session failed

Reply:
yeah, that try\catch can safely be removed since we check IsRunning before trying to start.  Only a race if someone else starts the session in the meantime (of course that would be an error we could safely ignore).

Doesn't really matter for this review since the entire "Create session" scenario isn't implemented and will be completely rewritten in the next PR today or tomorrow.